### PR TITLE
Time based parameters and attributes should use timedelta

### LINKED ISSUE #1223: nitclk needs to use timedelta
Several `nitclk` function parameters and attributes should be `timedelta` not float.
The function parameters should also have defaults.

   * finish_sync_pulse_sender_synchronize
   * setup_for_sync_pulse_sender_synchronize
   * synchronize
   * synchronize_to_sync_pulse_sender
   * wait_until_done
   * sample_clock_delay

### PR DESCRIPTION
**Pull request by texasaggie97**
- [X] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).
- [X] I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.
- [X] I've ~~added~~updated tests applicable for this pull request

### What does this Pull Request accomplish?
* Change time based parameters and attributes to optionally use `datetime.timedelta`
* Add default values

### List issues fixed by this Pull Request below, if any.
* Fix #1223 

### What testing has been done?
* Visual inspection of generated files
* Unit

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -74,6 +74,7 @@ All notable changes to this project will be documented in this file.
 * ### NI-TClk
     * #### Added
     * #### Changed
+        * Functions parameters and attributes that are time based now take or return a `datetime.timedelta` object
```
Comment (sbethur) [inline CHANGELOG.md]: "Method parameters and properties that are..."

File: src/nitclk/metadata/attributes.py [hunk 2]
```diff
@@ -110,6 +110,7 @@
     },
     11: {
         'access': 'read-write',
+        'attribute_class': 'AttributeViReal64TimeDeltaSeconds',
         'channel_based': False,
         'documentation': {
             'description': '\nSpecifies the sample clock delay.\nSpecifies the delay, in seconds, to apply to the session sample clock  relative to the other synchronized sessions. During synchronization,  NI-TClk aligns the sample clocks on the synchronized devices. If you want  to delay the sample clocks, set this attribute before calling  niTClk_Synchronize.\nnot supported for acquisition sessions.\nValues - Between minus one and plus one period of the sample clock.\nOne sample clock period is equal to (1/sample clock rate). For example,  for a session with sample rate of 100 MS/s, you can specify sample clock  delays between -10.0 ns and +10.0 ns.\nDefault Value is 0\n',
```
Comment (sbethur) [inline src/nitclk/metadata/attributes.py]: Documentation needs to be updated to mention about `timedelta`.

File: src/nitclk/metadata/functions.py [hunk 3]
```diff
@@ -134,11 +134,14 @@
             },
             {
                 'direction': 'in',
+                'default_value': 'datetime.timedelta(seconds=0.0)',
+                'python_api_converter_name': 'convert_timedelta_to_seconds',
+                'type_in_documentation': 'float in seconds or datetime.timedelta',
+                'type': 'ViReal64',
```
Comment (marcoskirsch) [inline src/nitclk/metadata/functions.py]: Please confirm these changes were made in internal NI metadata.

Comment (texasaggie97) [reply src/nitclk/metadata/functions.py]: They will be soon

File: CHANGELOG.md [hunk 4]
```diff
@@ -74,7 +74,7 @@ All notable changes to this project will be documented in this file.
 * ### NI-TClk
     * #### Added
     * #### Changed
-        * Functions parameters and attributes that are time based now take or return a `datetime.timedelta` object
+        * Mthod parameters and properties that are time based now take or return a `datetime.timedelta` object
```
Comment (marcoskirsch) [inline CHANGELOG.md]: Method (typo)

Comment (texasaggie97) [reply CHANGELOG.md]: Typing is hard...

Fixed

**Pull request merged**